test/integration/int_server.cc

```diff
@@ -0,0 +1,813 @@
+/* Copyright 2019 Istio Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "int_server.h"
+#include <future>
+#include "common/common/lock_guard.h"
+#include "common/common/logger.h"
+#include "common/grpc/codec.h"
+#include "common/http/conn_manager_config.h"
+#include "common/http/conn_manager_impl.h"
+#include "common/http/exception.h"
+#include "common/http/http1/codec_impl.h"
+#include "common/http/http2/codec_impl.h"
+#include "common/network/listen_socket_impl.h"
+#include "common/network/raw_buffer_socket.h"
+#include "envoy/http/codec.h"
+#include "envoy/network/transport_socket.h"
+#include "fmt/printf.h"
+#include "server/connection_handler_impl.h"
+#include "test/test_common/network_utility.h"
+#include "test/test_common/utility.h"
+
+namespace Mixer {
+namespace Integration {
+
+static Envoy::Http::LowerCaseString RequestId(std::string("x-request-id"));
+
+ServerStream::ServerStream() {}
+
+ServerStream::~ServerStream() {}
+
+class ServerStreamImpl : public ServerStream,
+                         public Envoy::Http::StreamDecoder,
+                         public Envoy::Http::StreamCallbacks,
+                         Envoy::Logger::Loggable<Envoy::Logger::Id::testing> {
+ public:
+  ServerStreamImpl(uint32_t id, ServerConnection &connection,
+                   ServerRequestCallback request_callback,
+                   Envoy::Http::StreamEncoder &stream_encoder)
+      : id_(id),
+        connection_(connection),
+        request_callback_(request_callback),
+        stream_encoder_(stream_encoder) {
+    // TODO do i have to do this? stream_encoder_.addCallbacks(*this);
```

Comment:
If so then you should have annotated all the overrides with "override".

Reply:
I'm pretty sure all overrides were annotated with 'override', but deleting the comment since its no longer relevant.  I added it before I had tested with http2 and was concerned that my working http1 code was going to break here for http2